Source/Library.cpp

```diff
@@ -10,31 +11,53 @@ Library::Library(QWidget *parent) :
 {
     ui->setupUi(this);
     this->setObjectName("libraryUI");
+    runningProcess = new QProcess(this);
+    processRunning = false;
+    connect(runningProcess, SIGNAL(finished(int, QProcess::ExitStatus)), this, SLOT(finished(int, QProcess::ExitStatus)));
+
 }
 
 Library::~Library()
 {
     delete ui;
+    delete runningProcess;
 }
 
 void Library::on_testLaunch_clicked()
 {
-    QFileDialog dialog(this);
-    dialog.setWindowTitle("Select Executable");
-    dialog.setFileMode(QFileDialog::ExistingFile);
-
-
-    if (dialog.exec()) {
-        QStringList files = dialog.selectedFiles();
-        QString file = files.at(0);
 
-        runProcess(file);
+    if (!processRunning) {
+        QFileDialog dialog();
```

Comment:
I'm getting build errors on both the initialization of the QFileDialog and the QMessageBox below, both in which the compiler thinks you're declaring a function "dialog()" with return type QFileDialog. You should specify the constructor by adding pointers to the parent widget in the initialization of each child.